overlord/servicestate/quota_control_test.go

```diff
@@ -78,16 +89,53 @@ type quotaGroupState struct {
 	Snaps          []string
 }
 
+func assertQuotaResources(c *C, grp *quota.Group, expected quota.Resources) {
+	if grp.MemoryLimit != 0 || expected.Memory != nil {
```

Comment:
All the `if`s in this test function make the test itself harder to validate. I'd suggest removing all of them, and just always compare the values.

In order to do this you'll need to make sure that the caller of this function always fills the `expected` structure in full, but I think it's worth it.

Reply:
Would this be ok instead? (I think I must have have forgotten about this)
```
	groupResources := grp.GetQuotaResources()
	c.Assert(groupResources, DeepEquals, expected)
```

Reply:
LGTM!